scripts/complete_liquidity_provision.js

```diff
@@ -81,42 +82,62 @@ const argv = default_yargs
+    let assertBracketsWellFormedPromise
+    let bracketsWithExistingOrdersPromise
+    let bracketAddresses
+    if (argv.brackets) {
+      bracketAddresses = argv.brackets
+      assertBracketsWellFormedPromise = verifyBracketsWellFormed(argv.masterSafe, bracketAddresses, null, null, true)
+      bracketsWithExistingOrdersPromise = exchangePromise.then((exchange) =>
+        findBracketsWithExistingOrders(bracketAddresses, exchange)
+      )
+    }
 
+    const exchange = await exchangePromise
     const tokenInfoPromises = fetchTokenInfoFromExchange(exchange, [argv.baseTokenId, argv.quoteTokenId])
     const baseTokenData = await tokenInfoPromises[argv.baseTokenId]
     const quoteTokenData = await tokenInfoPromises[argv.quoteTokenId]
     const { instance: baseToken, decimals: baseTokenDecimals } = baseTokenData
     const { instance: quoteToken, decimals: quoteTokenDecimals } = quoteTokenData
-
     const depositBaseToken = toErc20Units(argv.depositBaseToken, baseTokenDecimals)
     const depositQuoteToken = toErc20Units(argv.depositQuoteToken, quoteTokenDecimals)
+    const isBaseTokenBalanceSufficientPromise = checkSufficiencyOfBalance(baseToken, argv.masterSafe, depositBaseToken)
+    const isQuoteTokenBalanceSufficientPromise = checkSufficiencyOfBalance(quoteToken, argv.masterSafe, depositQuoteToken)
+    const priceCheckPromise = isPriceReasonable(baseTokenData, quoteTokenData, argv.currentPrice)
 
-    assert((await masterSafe.getOwners()).includes(signer), `Please ensure signer account ${signer} is an owner of masterSafe`)
+    const signer = await signerPromise
+    console.log("Using account:", signer)
+    assert((await masterOwnersPromise).includes(signer), `Please ensure signer account ${signer} is an owner of masterSafe`)
 
     if (argv.brackets) {
       assert(argv.numBrackets === argv.brackets.length, "Please ensure numBrackets equals number of brackets")
     }
 
     console.log("==> Performing safety checks")
-    if (!(await checkSufficiencyOfBalance(baseToken, masterSafe.address, depositBaseToken))) {
-      callback(`Error: MasterSafe ${masterSafe.address} has insufficient balance for base token ${baseToken.address}`)
+    if (!(await isBaseTokenBalanceSufficientPromise)) {
+      callback(`Error: MasterSafe ${argv.masterSafe} has insufficient balance for base token ${baseToken.address}`)
     }
-    if (!(await checkSufficiencyOfBalance(quoteToken, masterSafe.address, depositQuoteToken))) {
-      callback(`Error: MasterSafe ${masterSafe.address} has insufficient balance for quote token ${quoteToken.address}`)
+    if (!(await isQuoteTokenBalanceSufficientPromise)) {
+      callback(`Error: MasterSafe ${argv.masterSafe} has insufficient balance for quote token ${quoteToken.address}`)
     }
+
     // check price against dex.ag's API
-    const priceCheck = await isPriceReasonable(baseTokenData, quoteTokenData, argv.currentPrice)
-    if (!priceCheck) {
+    if (!(await priceCheckPromise)) {
```

Comment:
nit: although I am generally not a fan of having entire sentences for variable names, I feel like "priceCheck" has been neglected here. 

Also, if you are looking for something equivalent but slightly shorter for the "is" promises:

```
isXXXTokenBalanceSufficientPromise
hasSufficientXXXToken
```